pandas/core/style.py

```diff
@@ -0,0 +1,511 @@
+        a unique identifier to avoid CSS collisons; generated automatically
+    caption: str, default None
+        caption to attach to the table
+
+    Attributes
+    ----------
+    tempate: Jinja Template
+    ctx : defaultdict
+        maps (row, column) -> [styles] where each style is a
+        TODO: <(attribute, value)> paire | <string with 'attribute: value'>
+
+    Notes
+    -----
+    Most styling will be done through by passing style functions into
+    Styler.tee, Styler.apply, or Styler.applymap. Style functions should
+    return values with strings containing 'attr: value' that will be applied
+    to the indicated cells.
+
+    If using in the Jupyter notebook, Styler has defined a _repr_html_
+    to automatically render itself. Otherwise call Styler.render to get
+    the genterated HTML.
+    """
+    if has_jinja:
+        template = Template("""
+            <style  type="text/css" >
+            {% for s in table_styles %}
+                #T_{{uuid}} {{s.selector}} {
+                {% for p,val in s.props %}
+                    {{p}}: {{val}};
+                {% endfor %}
+                }
+            {% endfor %}
+            {% for s in cellstyle %}
+                #T_{{uuid}}{{s.selector}} {
+                {% for p,val in s.props %}
+                    {{p}}: {{val}};
+                {% endfor %}
+                }
+            {% endfor %}
+            </style>
+
+            <table id="T_{{uuid}}">
+            {% if caption %}
+                <caption>{{caption}}</caption>
+            {% endif %}
+
+            <thead>
+                {% for r in head %}
+                <tr>
+                    {% for c in r %}
+                    <{{c.type}} class="{{c.class}}">{{c.value}}
+                    {% endfor %}
+                </tr>
+                {% endfor %}
+            </thead>
+            <tbody>
+                {% for r in body %}
+                <tr>
+                    {% for c in r %}
+                    <{{c.type}} id="T_{{uuid}}{{c.id}}" class="{{c.class}}">
+                        {% if c.value is number %}
+                            {{c.value|round(precision)}}
+                        {% else %}
+                            {{c.value}}
+                        {% endif %}
+                    {% endfor %}
+                </tr>
+                {% endfor %}
+            </tbody>
+            </table>
+            """)
+
+    def __init__(self, data, precision=None, table_styles=None, uuid=None,
+                 caption=None):
+        self.ctx = defaultdict(list)
+
+        if data.ndim == 1:
+            data = data.to_frame()
+        if not data.index.is_unique or not data.columns.is_unique:
+            raise ValueError("Style is not supported for non-unique indicies.")
+
+        self.data = data
+        self.index = data.index
+        self.columns = data.columns
+
+        self.uuid = uuid
+        self.table_styles = table_styles
+        self.caption = caption
```

Comment:
ok though I think all of these should just be set_* methods 
makes things a lot simpler (and then work for both interactive and generated mode)

if u want to leave some kwargs in ok but then just call the appropriate method